lang/java/build.sh

```diff
@@ -44,6 +44,8 @@ function do_dist() {
 case "$target" in
   test)
     mvn -B test
+    # Test the modules that depend on hadoop using Hadoop 3
+    mvn -B test -Phadoop3 -pl 'mapred,tools,trevni'
```

Comment:
Side not: This can be error-prone if new modules depend on hadoop but I prefer to let it like is to not duplicate the test execution time.